cmd/clusterctl/cmd/get_kubeconfig_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func Test_intoKubeconfig(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	tt := []struct {
+		name     string
+		path     string
+		data     string
+		expected string
+	}{
+		{
+			name:     "inserting into empty kubeconfig",
+			path:     filepath.Join(tmpDir, "empty-kubeconfig"),
+			data:     "",
+			expected: workloadKubeconfig,
+		},
+		{
+			name:     "inserting into existing kubeconfig",
+			path:     filepath.Join(tmpDir, "existing-kubeconfig"),
+			data:     existingKubeconfig,
+			expected: joinedKubeconfig,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			if tc.data != "" {
+				err := os.WriteFile(tc.path, []byte(tc.data), 0600)
+				g.Expect(err).ToNot(HaveOccurred())
+			}
+
+			err := intoKubeconfig(tc.path, workloadKubeconfig)
+			g.Expect(err).ToNot(HaveOccurred())
+
+			expected, err := os.ReadFile(tc.path)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(string(expected)).To(Equal(tc.expected))
+		})
+	}
+}
+
+const workloadKubeconfig = `apiVersion: v1
```

Comment:
Let's separate to other file. We can embed these data to use go:embed.
It's more readable.

Reply:
Totally agree with you and embed is my preferable way for big data values in tests.
But I can see, that most of the test approaches in the Cluster API tests are using constants.

Reply:
Then, let's move constants to embedded way after it or asynchronously.

Reply:
Fixed, please review